web/src/connectors/task-annotator-connector/use-document-data-lazy-loading/index.ts

```diff
@@ -0,0 +1,114 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useAnnotationsLazyLoading } from './use-annotations-lazy-loading';
+import { useTokensLazyLoading } from './use-tokens-lazy-loading';
+import { TDocumentDataLazyLoadingParams } from './types';
+
+const createPendingPromise = () => {
```

Comment:
This helper is needed to help signalise that all needed data is loaded at some moment (`InfiniteLoader` component requires to return resolved promise in `loadMoreItems` callback when all data is loaded).